python/pyarrow/tests/test_table.py

```diff
@@ -1330,6 +1330,22 @@ def test_concat_tables():
     assert result.equals(expected)
 
 
+def test_concat_tables_permissive():
+    t1 = pa.Table.from_arrays([list(range(10))], names=('a',))
+    t2 = pa.Table.from_arrays([list(range(10, 20))], names=('a',))
+
+    result = pa.concat_tables([t1, t2], field_merge_options="permissive")
```

Comment:
I am not sure this is a perfect test for "permissive", because this specific case would perfectly work as well without specifying `field_merge_options="permissive"` (since the types are the same), and so it doesn't test that the option is properly passed through

Reply:
Fair point, updated it to a rainy day situation